src/main/scala/com/databricks/spark/xml/XmlInputFormat.scala

```diff
@@ -164,7 +164,7 @@ private[xml] class XmlRecordReader extends RecordReader[LongWritable, Text] {
         // End of file or end of split.
         return false
       } else {
-        if (b == startTag(i)) {
+        if (b.toByte == startTag(i)) {
```

Comment:
Sorry, @neherim, mind elaborating how it fixes the issue?

Reply:
Of course, described fix in comments.